go/roothash/api/message/message.go

```diff
@@ -42,3 +47,25 @@ type NoopMessage struct {
 func (nm *NoopMessage) ValidateBasic() error {
 	return nil
 }
+
+// StakingMessage is a runtime message that allows a runtime to perform staking operations.
+type StakingMessage struct {
+	cbor.Versioned
+
+	Transfer *staking.Transfer `json:"transfer,omitempty"`
+	Withdraw *staking.Withdraw `json:"withdraw,omitempty"`
+}
+
+// ValidateBasic performs basic validation of the runtime message.
+func (sm *StakingMessage) ValidateBasic() error {
```

Comment:
Should this also validate that both `Transfer` & `Withdraw` are not set? (since this is assumed by rest of the code).

Reply:
Yeah, done.